compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/Fir2IrConversionScope.kt

```diff
@@ -77,15 +77,17 @@ class Fir2IrConversionScope {
     fun parent(): IrDeclarationParent? = parentStack.lastOrNull()
 
     fun lastDispatchReceiverParameter(): IrValueParameter? {
-        val dispatchReceiver = functionStack.lastOrNull()?.dispatchReceiverParameter
-        return if (dispatchReceiver != null) {
-            // Use the dispatch receiver of the containing function
-            dispatchReceiver
-        } else {
-            val lastClassSymbol = classStack.lastOrNull()?.symbol
-            // Use the dispatch receiver of the containing class
-            lastClassSymbol?.owner?.thisReceiver
-        }
+        var dispatchReceiver = functionStack.lastOrNull()?.dispatchReceiverParameter
+        // Use the dispatch receiver of the containing function
+        dispatchReceiver?.let { return it }
+
+        // Use the dispatch receiver of the enclosing function if any
+        dispatchReceiver = functionStack.elementAtOrNull(functionStack.size - 2)?.dispatchReceiverParameter
+        dispatchReceiver?.let { return it }
```

Comment:
What if there is more than one nested anonymous function, like:
```
class My {
    val my: String = "O"
        get() = { { field }() }() + "K"
}
```
I'm just wondering if you should keep going up the stack to find a receiver (i.e., with a while loop).

Reply:
Thanks for the suggestion! I'll try that one as well, and see if we actually need to look up function stack all the way up.

Reply:
Indeed, having more nested lambdas failed on FIR, but succeed on IR. Thanks for the suggestion again!